### PR TITLE
chore(go): upgrade go 1.21.7

### PR DESCRIPTION
**Pull request by tranngoclam**
## Description

- Upgrade go to 1.21.7 and base chainguard static image

## References

https://github.com/golang/go/issues?q=milestone%3AGo1.21.7+label%3ACherryPickApproved

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openfga/openfga
 
-go 1.21.5
+go 1.22.0
```
Comment (rhamzeh) [inline go.mod:3]: Can you please keep this at the latest go1.21 version? Currently `go1.21.7`

File: tools/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/openfga/openfga/tooling
 
-go 1.21
+go 1.22.0
```
Comment (rhamzeh) [inline tools/go.mod:3]: Can you please keep this at the latest go1.21 version? Currently `go1.21.7`

**Review by rhamzeh — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (tranngoclam) [reply go.mod]: I have updated to go 1.21.7

File: tools/go.mod (hunk 2 above)
Comment (tranngoclam) [reply tools/go.mod]: I have updated to go 1.21.7

**Review by rhamzeh — DISMISSED** [review]

File: .golangci.yaml [hunk 3]
```diff
@@ -46,3 +46,7 @@ linters-settings:
     alias:
       - pkg: github.com/openfga/api/proto/openfga/v1
         alias: openfgav1
+  testifylint:
+    enable-all: true
+    disable:
+      - go-require
```
Comment (miparnisari) [inline .golangci.yaml:52]: Please undo this change, it's been fixed :)

**Review by miparnisari — CHANGES_REQUESTED** [review]

**Review by miparnisari — APPROVED** [review]
